### PR TITLE
fundamental-react 0.5.1

### PR DESCRIPTION
**Pull request by Feinbube**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/fundamental-react/lib/Tabs.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    /* _INTERNAL USE ONLY._ */
+    selected?: boolean;
+    /* Additional props to be spread to the tab content's <div> element. */
+    tabContentProps?: { [x: string]: any };
+    /* Localized text to display on the tab. */
+    title?: string;
+    /* _INTERNAL USE ONLY._ */
+    onClick?: (
+        event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+        index?: number
```
Comment (PranavSenthilnathan) [inline types/fundamental-react/lib/Tabs.d.ts]: As discussed [here](http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html), it's not recommended to have optional parameters in callbacks. The [underlying library](https://github.com/SAP/fundamental-react/blob/master/src/Tabs/Tab.js#L44) seems to always call this function with an `index` argument.

File: types/fundamental-react/lib/ActionBar.d.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    /* Localized text for the description. */
+    description?: string;
+    /* Additional props to be spread to the description's `<p>` element. */
+    descriptionProps?: { [x: string]: any };
+    /* Heading level. `<h1>` is reserved for the page title. */
+    headingLevel?: 1 | 2 | 3 | 4 | 5 | 6;
+    titleProps?: { [x: string]: any };
+} & { [x: string]: any };
+
+export class ActionBar extends React.Component<ActionBarProps> {
```
Comment (PranavSenthilnathan) [inline types/fundamental-react/lib/ActionBar.d.ts]: It looks like `fundamental-react` organizes its internal exports as `default` exports and then exports them in the index.js with different names. this code exports with the final name and then the index.d.ts just exports without renaming. There's no functional difference between the two, but if you wanted you could use default exports here like the original library.

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

File: types/fundamental-react/lib/Tabs.d.ts (hunk 1 above)
Comment (Feinbube) [reply types/fundamental-react/lib/Tabs.d.ts]: that's great feedback! thank you :)

File: types/fundamental-react/lib/ActionBar.d.ts (hunk 2 above)
Comment (Feinbube) [reply types/fundamental-react/lib/ActionBar.d.ts]: Thank you for your feedback. It turned out that it did make a difference in this project. I reworked the folder structure and covered all the other ways to import classes from fundamental-react.

Comment (Feinbube) [reply types/fundamental-react/lib/ActionBar.d.ts]:
```
import { FormSet } from "fundamental-react";
import { FormItem } from "fundamental-react/lib";
import { FormLabel } from "fundamental-react/lib/Forms";
import FormInput from "fundamental-react/lib/Forms/FormInput";
```

**Review by PranavSenthilnathan — APPROVED** [review]

**Comment by PranavSenthilnathan** [comment]
Congratulations on your first DefinitelyTyped contribution! Thank you for being a part of the community!

**Pull request merged**